camviewer_ui_impl.py

```diff
@@ -1151,30 +1151,58 @@ def shutdown(self):
 
     def onfileSave(self):
         try:
-            fileName = QFileDialog.getSaveFileName(
+            filename = QFileDialog.getSaveFileName(
                 self,
                 "Save Image...",
-                self.cwd,
+                os.path.expanduser("~"),
                 "Images (*.npy *.jpg *.png *.bmp *.pgm *.tif)",
             )[0]
-            if fileName == "":
+            if filename == "":
                 raise Exception("No File Name Specified")
-            if fileName.lower().endswith(".npy"):
-                np.save(fileName, self.image)
+            if filename.lower().endswith(".npy"):
+                np.save(filename, self.image)
                 QMessageBox.information(
                     self,
                     "File Save Succeeded",
-                    "Image has been saved as a numpy file: %s" % (fileName),
+                    "Image has been saved as a numpy file: %s" % (filename),
                 )
-                print("Saved to a numpy file %s" % (fileName))
-            else:
-                self.ui.display_image.image.save(fileName, format=None, quality=-1)
+                print("Saved to a numpy file %s" % (filename))
+            elif self.ui.display_image.image.save(filename, format=None, quality=-1):
```

Comment:
It's weird to me to see expressions with side-effects in conditional arguments.  I think it's a code smell thing primarily, but it's also readable enough.  

Refactoring would probably just be pulling the `image.save` call out and starting a new `if else` block

Reply:
I agree, I probably shouldn't have combined these together, if only for readability. It's pretty easy to get confused reading this as-is.

Reply:
Yep, agree that side effects in conditional statements aren't ideal. I might just move `print("Saved to a numpy file %s" % (filename))` outside of the if-else statement.

Reply:
Resolved by refactoring. It should now be clear how this is being used.